use_cases/eluc/prescriptors/nsga2/candidate.py

```diff
@@ -0,0 +1,95 @@
+"""
+Contains the Candidate class, which is a simple feed-forward neural network that keeps track
+of its own metrics and logging information.
+"""
+import torch
+
+class Candidate(torch.nn.Module):
+    """
+    Simple fixed topology 1 hidden layer feed-forward nn candidate.
+    Keeps track of its own metrics and evolution logging information.
+    """
+    def __init__(self, in_size: int, hidden_size: int, out_size: int, device="cpu", gen=-1, cand_id=-1, parents=(None, None)):
+        super().__init__()
+
+        self.in_size = in_size
+        self.hidden_size = hidden_size
+        self.out_size = out_size
+
+        self.model = torch.nn.Sequential(
+            torch.nn.Linear(in_size, hidden_size),
+            torch.nn.Tanh(),
+            torch.nn.Linear(hidden_size, out_size))
+        
+        self.device = device
+        self.model.to(device)
+        self.model.eval()
+        
+        # Orthogonal initialization
+        for layer in self.model:
+            if isinstance(layer, torch.nn.Linear):
+                torch.nn.init.orthogonal_(layer.weight)
+                layer.bias.data.fill_(0.01)
+
+        # To keep track of metrics
+        self.metrics = None
+        self.rank = None
+        self.distance = -1
```

Comment:
We keep track of our own metrics rather than having them kept in some external dict in the NSGA-II algorithm code.